### ISSUE TITLE
Option to Disable Auto-Space After Suggestions (Except After Punctuation)

### ISSUE THREAD

**Issue by Erron-26** [futo-org/android-keyboard#765]
### Use Cases

Improving typing accuracy by avoiding unwanted spaces after suggestions, especially useful for users experiencing issues where spaces are inserted on the wrong side (? 

### Proposal

An option to disable spaces after inserting a suggestion, but still keep spaces after punctuation would be great. Sometimes that auto-space feature after a suggestion adds these extra spaces out of nowhere, especially because that bug where spaces get added to the left instead of the right is happening again in this official version. So until those little issues are fully fixed, an option like this would really help. In the end, the more options, the better the app will be

**Comment by dean-futo** [comment]
I'm assuming you've upgraded and are using 0.1.24? Will you please upload a video of the extra space bug? It will likely be a matter of fixing the bug rather than implementing a feature, as adding a space after suggestions is standard.

**Comment by Erron-26** [comment]
Yeah, I'm on the latest official version. I was going to say I have no idea why this bug is happening because, unlike last time I reported it, it's gotten a lot harder to catch (since the recent fixes have improved the keyboard). But I think I just found a specific case where this bug shows up:

* I type a word, the auto-correct fixes it, but that’s not the word I actually wanted, so I hit backspace to undo the correction. I check the suggestions, see the word I actually wanted, select it, and bam—"double space" gets added.

https://github.com/user-attachments/assets/b5b1a4e7-85c4-4373-b9a1-8e0fa7a8aead

**Comment by Erron-26** [comment]
It might seem like a small thing in this isolated case, but imagine typing fast—you might not even notice it until after you’ve sent the message

**Comment by abb128** [comment]
The bug you recorded will be fixed in the next minor release. If you find any other bugs you can let me know

**Comment by Erron-26** [comment]
Hi @abb128, I have a question: in the last commits, was this bug fixed, or are you talking about fixing it in the next stable release? I wouldn't see with bad eyes to test the nightly build once again even if the current stable release is not bad, but you know, this problem sometimes is really annoying

**Comment by Erron-26** [comment]
Hey, I’ve got some good news:

In version 0.1.24.1-rc1-playstore, this issue seems partially fixed when writing more complete sentences. However, I’m still seeing that when you type a single word, like `hola`, and correct it (like in the video I shared earlier), the double space still gets inserted.

Also, I’m not sure if this is intentional or a bug, but the `;` no longer adds an automatic space in Spanish, even though it’s a punctuation mark like the `.`. I don’t really get why there’s a difference between them

**Comment by Erron-26** [comment]
If you need a video showing what I just mentioned, just let me know

**Comment by Erron-26** [comment]
This bug keeps occurring in version 0.1.24.2-rc1-playstore. The latest improvements have made the keyboard much more solid for day-to-day use, but these small issues can greatly damage the daily experience. Especially this bug that, now with the multilingual corrections, has been appearing more frequently for me

I think you should take a look at it before the next release candidate

https://github.com/user-attachments/assets/7bab8b37-c420-4ea8-a5e2-eebd963a348c

**Comment by codethief** [comment]
The issue is still occurring for me in version 0.1.24.2-playstore.

Together with https://github.com/futo-org/android-keyboard/issues/815, auto-spacing just seems completely broken and unintuitive to me. I have finally gotten tired of it and just disabled the feature altogether.
